### ISSUE TITLE
Add support for running tests in local desktop via build/bundle manifest

### ISSUE THREAD

**Issue by saratvemulapalli** [opensearch-project/opensearch-build#629]
### Is your feature request related to a problem? Please describe

All test suites depend on pulling the manifest from S3. This experience is not great to run tests when manifests are present locally.

Coming from conversation of: https://github.com/opensearch-project/project-website/pull/293

### Describe the solution you'd like

Run test suites taking manifest as parameters and store test results in local.
```
./bundle-workflow/test.sh manifests/opensearch-1.1.0.yml
```

**Comment by setiah** [comment]
The test-workflow depends on the directory layout for artifacts set by build workflow defines, to read below. 
* Maven artifacts
* Built zip dependencies like job-scheduler
* Bundles artifacts.

test-workflow is able to generate the download paths using `s3 bucket`, `build_id`, `architecture`. if we want to build and test locally, we might need to modify the inputs to pre-provided paths. These paths could be s3 or local or remote url. Doable but interface seems less cleaner than what it is. The interfacing job between the build workflow and test workflow will have to generate these paths for a build id to invoke test workflow. 

Another challenge with running test workflow locally is that the perf-test require vpc peering with rally setup, which is outside of the test setup process. This means we could enable only the integ and bwc test workflows to run locally

**Comment by peterzhuamazon** [comment]
Please provide more information on how much work we need to put into this.

**Comment by dblock** [comment]
I have implemented downloading the manifest and exporting all the files from a distribution using publicly available URLs in https://github.com/opensearch-project/opensearch-build/pull/789. After that's merged (cc: @saratvemulapalli and @peternied) I think we can replace all `from_s3` by `from_url`, followed by a new `.download()` that's a subset of the `export_artifacts()` method I added. I think there's a day's worth of work here.
